src/ngraph/runtime/cpu/mkldnn_emitter.hpp

```diff
@@ -823,6 +1278,166 @@ namespace ngraph
                     m_mkldnn_primitives[conv_idx] = prim;
                 }
 
+                template <typename OP>
+                mkldnn::convolution_backward_data::desc
+                    get_convolution_backward_data_desc(const ngraph::Node* node)
+                {
+                    auto convolution = static_cast<const OP*>(node);
+                    // For dilation, MKLDNN wants to know how many elements to insert between, not how far
+                    // apart to space the elements like nGraph. So we have to subtract 1 from each pos.
+                    Strides window_dilation_strides_adjusted;
+
+                    for (size_t s : convolution->get_window_dilation_strides_forward())
+                    {
+                        window_dilation_strides_adjusted.push_back(s - 1);
+                    }
+
+                    auto weights_desc = mkldnn_utils::get_input_mkldnn_md(node, 0);
+                    // MKLDNN relies on named formats for kernel selection
+                    if (weights_desc.data.format == mkldnn_nchw)
+                        weights_desc.data.format = mkldnn_oihw;
+                    if (weights_desc.data.format == mkldnn_ncdhw)
+                        weights_desc.data.format = mkldnn_oidhw;
```

Comment:
should we use curly braces here, according to coding standard or single line then/else/for are ok without them?